src/hsmm_mvpy/models.py

```diff
@@ -926,30 +919,27 @@ def estim_probs(self, magnitudes, parameters, n_events=None, subset_epochs=None,
                 subset_epochs = np.where(subset_epochs)[0]
             n_trials = len(subset_epochs)
             durations = self.durations[subset_epochs]
-            starts = self.starts[subset_epochs]
-            ends = self.ends[subset_epochs]
         else:
             n_trials = self.n_trials
             durations = self.durations
-            starts = self.starts
-            ends = self.ends
 
-        gains = np.zeros((self.n_samples, n_events), dtype=np.float64)
+        gains = np.zeros((self.max_d, n_trials, n_events), dtype=np.float64)
         for i in range(self.n_dims):
             # computes the gains, i.e. congruence between the pattern shape
             # and the data given the magnitudes of the sensors
-            gains = gains + self.events[:,i][np.newaxis].T * magnitudes[:,i]-magnitudes[:,i]**2/2
+            for tr in range(n_trials):
+                gains[:, tr, :] =  gains[:, tr, :] + self.events[:,tr,i][np.newaxis].T * magnitudes[:,i]-magnitudes[:,i]**2/2
+        
         gains = np.exp(gains)
         probs = np.zeros([self.max_d, n_trials,n_events], dtype=np.float64) # prob per trial
         probs_b = np.zeros([self.max_d, n_trials,n_events], dtype=np.float64)# Sample and state reversed
         for trial in np.arange(n_trials):
```

Comment:
This part can then be rewritten, I can take a look this afternoon

Reply:
Yeah of course, that makes sense. Even better :)